matheval.class.php

```diff
@@ -214,7 +214,14 @@ protected function nfx($expr) {
                     if (is_null($o2)) return $this->trigger("unexpected ')'");
                     else $output[] = $o2;
                 }
-                if (preg_match("/^([a-z]\w*)\($/", $stack->last(2), $matches)) { // did we just close a function?
+                
+                // apply in case php version >= 8.1.0: apply a null coalescing operator if stack->last(2) does not exist or is null
+                $stacklast2 = $stack->last(2);
+                if (version_compare(PHP_VERSION, '8.1.0') >= 0) {                
```

Comment:
Instead of checking for the PHP version, would it be easier to just do this?

`$stacklast2 = !empty( $stack->last(2)) ? $stack->last(2) : '';`

Reply:
Yes, I was thinking the same.